tools/generate-wire.py

```diff
@@ -873,7 +874,7 @@ def print_struct(self):
 \t\t\t*plen -= msg_len;
 \t\t}}
 \t}}
-\tif (!*p) {{
+\tif (!*p || start_len - *plen != *len) {{
```

Comment:
It's not entirely clear to me that this is correct.  For onion, at least, we should on first type=0 and ignore any remaining.  Seems consistent to do that elsewhere, too.

Reply:
hmm you're right. there's also the 'ordering' aspect of the proposed [TLV spec](https://github.com/lightningnetwork/lightning-rfc/pull/572) that this implementation is missing at the moment. will fix in a second PR!